### ISSUE TITLE
Adding keywords to language at runtime

### ISSUE THREAD

**Issue by erikmellum** [isagalaev/highlight.js#1271]
I have a set of keywords that need to be added to my mathematica language definition at runtime. The reason I can't just add these into the source is that they get updated regularly. Is there a way to dynamically add the keywords at runtime?

**Comment by isagalaev** [comment]
Sorry for the delayed reply. You could do something like this:
```js
hljs.getLanguage('mathematica').keywords += ' keyword1 keyword2'; // note the leading space
```
Make sure you do it before the first attempt at highlighting anything as the keywords get compiled into a more efficient runtime structure at the first use of a language.

**Comment by isagalaev** [comment]
Oh, and if it's a compressed browser version, use `.k` instead of `.keywords`.

**Comment by erikmellum** [comment]
Thank you for the response 👍 . I will switch to do it this way. Right now I am doing:

```
hljs.highlightBlock(document.createElement("pre"));
var obj = hljs.getLanguage("mathematica").k;
keywords.forEach(function(value) {
obj[value] = ["keyword", 1];
});
```

I noticed that hljs changed the internal structures when the first highlight attempt occurs. I think I first tried to add my keywords before the first highlight attempt, and wasn't having success. I probably just wasn't making sure that the keywords were added before the first highlight attempt. At any rate, I added them this way which works, but is definitely hackier since I don't fully know how the array in the value is used. I'm assuming index 0 in the array just helps detect the style to apply, but I'm not sure what the number at index 1 is for.
